src/wilcoxon.jl

```diff
@@ -163,6 +163,26 @@ function StatsAPI.pvalue(x::ExactSignedRankTest; tail=:both)
     end
 end
 
+function psignrank(W::T, n::T, lower_tail::Bool) where {T<:Integer}
+    max_W =  n * (n + one(T)) ÷ (one(T) + one(T))
+    if !lower_tail
+        return psignrank(max_W - W, n, !lower_tail)
+    end
+    if W == max_W
+        return 1.0
```

Comment:
It seems this already assumes we only work with `Int` and `Float64`? Otherwise it would cause a type instability.